Jenkins_jobs/WorkspaceInfo.groovy

```diff
@@ -1,138 +1,100 @@
 @Library('NodeHelper') _
 import jenkins.model.Jenkins;
 import hudson.model.Computer;
+import org.jenkinsci.plugins.workflow.steps.TimeoutStep;
 
 clones = [:]
-List<String> newMachines = new ArrayList<String>();
 
 /* Iterates over all the online nodes in Jenkins and
  * prints contents of workspace folder along with
  * the space they occupy. With exception of tmp directory
+ * 
+ * The nodes can be either passed in or grabbed from Jenkins
  */
 node {
-    stage('Print_Space_Monitoring_Data') {
+    stage('Pepare_Space_Monitoring_Jobs') {
         NodeHelper nodeHelper = new NodeHelper();
 
-        String projectLabel;
+        String projectLabel
         if (params.projectLabel.length() > 1) {
-            projectLabel = params.projectLabel;            
+            projectLabel = params.projectLabel
         } else {
-            projectLabel = "all";
+            projectLabel = "all"
         }
-
-        def currentInstance = Jenkins.getInstance();
-        Computer[] computers;
-
-        if (currentInstance != null) {
-            computers = currentInstance.getComputers();
+        def timeOut = new TimeoutStep(Integer.parseInt(params.timeout))
+
+        
+        ArrayList<Computer> computers = new ArrayList<Computer>()
+        if (params.machines.length() > 1) {
+            String[] inputMachines = params.machines.split(",")
+            for (String inputMachine : inputMachines) {
+                computers.add(nodeHelper.getComputer(inputMachine))
+            }
+        } else {
+            def currentInstance = Jenkins.getInstance()
+            if (currentInstance != null) {
+                computers = currentInstance.getComputers();
+            }
         }
 
+
         for (Computer computer : computers) {
             String machineName = computer.getName();
 
             if (computer.isOnline() && computer.getName() != "") {
```

Comment:
`computer.getName()` already stored in `machineName`

Reply:
This is in case `machineName` has the domain but the computer on Jenkins doesn't have the domain in it's name. If I use `machineName` it won't work. Hence, `computer.getName()`

Reply:
I don't understand
```
machineName = computer.getName()
```
So why have `if computer.getName()` when you already have it in a variable?

Reply:
ah yes, you are right that is redundant. I completely missed the first if condition which validates the computer names. I'll take it out